### PR TITLE
Handle empty/unknown metrics status for HPAs

### PR DESCRIPTION
**Pull request by aperkins1310**
When all pods in a deployment are restarting, the CPU utilisation metric can't be computed, and therefore k8s puts an "empty" type in the status:

```yaml
apiVersion: autoscaling/v2
kind: HorizontalPodAutoscaler
metadata:
  creationTimestamp: "2024-02-20T16:16:13Z"
  labels:
  name: my-hpa
  namespace: my-ns
  resourceVersion: "3047377695"
  uid: 54b0a87a-63c7-4d81-aca1-9758f40d23af
spec:
  maxReplicas: 10
  metrics:
  - resource:
      name: cpu
      target:
        averageUtilization: 70
        type: Utilization
    type: Resource
  minReplicas: 1
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: my-app
status:
  conditions:
  - lastTransitionTime: "2024-02-20T16:16:29Z"
    message: recommended size matches current size
    reason: ReadyForNewScale
    status: "True"
    type: AbleToScale
  - lastTransitionTime: "2024-02-20T16:16:29Z"
    message: 'the HPA was unable to compute the replica count: No recommendation'
    reason: FailedGetResourceMetric
    status: "False"
    type: ScalingActive
  - lastTransitionTime: "2024-02-20T16:16:29Z"
    message: the desired count is within the acceptable range
    reason: DesiredWithinRange
    status: "False"
    type: ScalingLimited
  currentMetrics:
  - type: ""
  currentReplicas: 1
  desiredReplicas: 1
```
currently Skuber fails with a match error:
```
scala.MatchError:  (of class java.lang.String)
	at skuber.autoscaling.v2.HorizontalPodAutoscaler$$anon$3.reads(HorizontalPodAutoscaler.scala:254)
	at play.api.libs.json.Format$$anon$6.reads(Format.scala:78)
	at play.api.libs.json.LowPriorityDefaultReads$$anon$8.$anonfun$reads$1(Reads.scala:289)
	at scala.collection.IterableOnceOps.foldLeft(IterableOnce.scala:676)
	at scala.collection.IterableOnceOps.foldLeft$(IterableOnce.scala:670)
	at scala.collection.AbstractIterator.foldLeft(Iterator.scala:1300)
	at play.api.libs.json.LowPriorityDefaultReads$$anon$8.reads(Reads.scala:288)
	at play.api.libs.json.DefaultFormat$$anon$7.reads(Format.scala:88)
	at play.api.libs.json.PathReads.$anonfun$nullableWithDefault$1(JsConstraints.scala:64)
	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:244)
	at play.api.libs.json.OFormat$$anon$4.reads(Format.scala:58)
	at play.api.libs.json.OFormat$$anon$2.$anonfun$inmap$1(Format.scala:43)
	at play.api.libs.json.OFormat$$anon$3.reads(Format.scala:52)
	at play.api.libs.json.Reads$$anon$1$$anon$2.reads(Reads.scala:219)
	at play.api.libs.json.OFormat$$anon$4.reads(Format.scala:58)
	at play.api.libs.json.Reads.$anonfun$map$1(Reads.scala:48)
	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:244)
	at play.api.libs.json.Reads$$anon$1$$anon$2.reads(Reads.scala:219)
	at play.api.libs.json.OFormat$$anon$4.reads(Format.scala:58)
	at play.api.libs.json.OFormat$$anon$2.$anonfun$inmap$1(Format.scala:43)
	at play.api.libs.json.OFormat$$anon$3.reads(Format.scala:52)
	at play.api.libs.json.PathReads.$anonfun$nullableWithDefault$1(JsConstraints.scala:64)
	at play.api.libs.json.Reads$$anon$6.reads(Reads.scala:244)
	at play.api.libs.json.OFormat$$anon$4.reads(Format.scala:58)
	at play.api.libs.json.Reads$$anon$1$$anon$2.reads(Reads.scala:219)
	at play.api.libs.json.OFormat$$anon$4.reads(Format.scala:58)
	at play.api.libs.json.OFormat$$anon$2.$anonfun$inmap$1(Format.scala:43)
	at play.api.libs.json.OFormat$$anon$3.reads(Format.scala:52)
	at play.api.libs.json.LowPriorityDefaultReads$$anon$8.$anonfun$reads$1(Reads.scala:289)
	at scala.collection.IterableOnceOps.foldLeft(IterableOnce.scala:676)
	at scala.collection.IterableOnceOps.foldLeft$(IterableOnce.scala:670)
	at scala.collection.AbstractIterator.foldLeft(Iterator.scala:1300)
	at play.api.libs.json.LowPriorityDefaultReads$$anon$8.reads(Reads.scala:288)
	at play.api.libs.json.DefaultFormat$$anon$7.reads(Format.scala:88)
```

### CODE REVIEW

File: client/src/main/scala/skuber/autoscaling/v2/HorizontalPodAutoscaler.scala [hunk 1]
```diff
@@ -257,6 +263,7 @@ object HorizontalPodAutoscaler {
         case "RESOURCE" => JsSuccess(json.as[ResourceMetricStatusHolder])
         case "EXTERNAL" => JsSuccess(json.as[ExternalMetricStatusHolder])
         case "CONTAINERRESOURCE" => JsSuccess(json.as[ContainerResourceMetricStatusHolder])
+        case "" => JsSuccess(EmptyMetricStatus)
```
Comment (hagay3) [inline client/src/main/scala/skuber/autoscaling/v2/HorizontalPodAutoscaler.scala]: lets make it more resilient and replace
```
case "" => JsSuccess(EmptyMetricStatus)
```
with
```
case _ => JsSuccess(EmptyMetricStatus)
```
wdyt ?

**Review by hagay3 — CHANGES_REQUESTED** [review]
resiliency

**Comment by hagay3** [comment]
@aperkins1310 
Thank you for taking the time and contributing to skuber :)

File: client/src/main/scala/skuber/autoscaling/v2/HorizontalPodAutoscaler.scala (hunk 1 above)
Comment (aperkins1310) [reply client/src/main/scala/skuber/autoscaling/v2/HorizontalPodAutoscaler.scala]: makes sense to me, although maybe then it should be `UnknownMetricStatus`?

Comment (hagay3) [reply client/src/main/scala/skuber/autoscaling/v2/HorizontalPodAutoscaler.scala]: sure sounds good

**Review by hagay3 — APPROVED** [review]

**Pull request merged**